packages/ui/src/services/breakpoints/plugin/create-service.ts

```diff
@@ -0,0 +1,143 @@
+import { App, Ref, computed, watch, reactive, ComputedRef } from 'vue'
+
+import { useEvent } from '../../../composables'
+
+import { warn, generateUniqueId } from '../../utils'
+import { isClient } from '../../../utils/ssr-utils'
+import { getGlobalProperty } from '../../../vuestic-plugin/utils'
+import { addOrUpdateStyleElement } from '../../dom-functions'
+
+import { GlobalConfig } from '../../global-config/types'
+import { ThresholdsKeys, BreakpointsConfig, BodyClass, WindowSizes } from '../types'
+
+export const createBreakpointsConfigPlugin = (app: App) => {
+  const globalConfig: Ref<GlobalConfig> | undefined = getGlobalProperty(app, '$vaConfig')?.globalConfig
+  if (!globalConfig) {
+    warn('createBreakpointsConfigPlugin: globalConfig is not defined!')
+    return {}
+  }
+
+  const breakpointsConfig: ComputedRef<BreakpointsConfig> = computed(() => {
+    const breakpoints = globalConfig.value.breakpoints
+    if (!breakpoints) { warn('createBreakpointsConfigPlugin: breakpointsConfig is not defined!') }
+    return breakpoints ?? {} as BreakpointsConfig
+  })
+
+  if (!breakpointsConfig.value.enabled) {
+    return {}
+  }
+
+  if (!breakpointsConfig.value.thresholds || !Object.values(breakpointsConfig.value.thresholds).length) {
+    warn('createBreakpointsConfigPlugin: there are no defined thresholds!')
+    return {}
+  }
+
+  const isMounted = computed(isClient)
+  const windowSizes = reactive<WindowSizes>({
+    width: undefined,
+    height: undefined,
+  })
+
+  const setCurrentWindowSizes = () => {
+    windowSizes.width = window?.innerWidth
+    windowSizes.height = window?.innerHeight
+  }
+
+  watch(isMounted, (v) => {
+    if (!v) { return }
+    setCurrentWindowSizes()
+    useEvent('resize', setCurrentWindowSizes, true)
+  }, { immediate: true })
+
+  const currentBreakpoint = computed(() => {
+    if (!isMounted.value || !windowSizes.width) { return }
+
+    return Object.entries(breakpointsConfig.value.thresholds).reduce((acc: string, [key, value]) => {
+      if (windowSizes.width! >= value) { acc = key }
+      return acc
+    }, 'xs') as unknown as ThresholdsKeys
```

Comment:
It's not an array or object, so it must be singular.

```suggestion
    }, 'xs') as unknown as ThresholdsKey
```